garak/generators/huggingface.py

```diff
@@ -512,14 +549,22 @@ def _call_model(
                         return returnval
                     else:
                         raise e
-                text_output = self.tokenizer.batch_decode(
+                raw_text_output = self.tokenizer.batch_decode(
                     outputs, skip_special_tokens=True, device=self.device
                 )
 
+        if self.use_chat:
+            text_output = [
+                re.sub("^" + re.escape(prefix_prompt), "", i).strip()
+                for i in raw_text_output
+            ]
+        else:
+            text_output = raw_text_output
```

Comment:
Part of me (albeit a part with limited convictions) feels like there HAS to be a better way to handle this. HF seems to be managing it internally with their `pipeline`s, but this seems fine for the time being.

Reply:
idk man the format uses `\n` as a special char but doesn't seem to do mush escaping, many bets are off

Another option could be to do `\n`-based escaping and parsing of messages line by line, assuming `[A-Z][a-z]+:` is a metadata line

My experience also is that things like `prefix_prompt` get mangled in other code sometimes. I don't have great suggestions for how to check this well, other than:
* continue doing qualitative reviews of prompt:output pairs for suspicious items;
* migrate to a data format with clearer separation between data and metadata;
* badger HF for a reference implementation;
* give notice to users that HF Chat is "janky".